### DISCUSSION TITLE
lookup custom formula

### DISCUSSION THREAD

**Discussion by Rhydderch** [mdelobelle/metadatamenu#592] [Q&A]
Hi,

I have a list of features notes which have lookup fields containing features they depend on.

I can show the number of incomplete features they depend on with the following dataviewjs code

```dataviewjs
const currentPage = dv.current(); // This retrieves the current page's data.
let countNotDone = 0;

if (currentPage.depends_on_features) {
    for (const feature of currentPage.depends_on_features) {
        // Assuming 'feature' is a link to another note.
        const featurePage = dv.page(feature);
        // Check if the feature_status is not "Completed".
        if (featurePage && featurePage.feature_status !== "Completed") {
            countNotDone++;
        }
    }
}

// Output the count of not completed features.
dv.paragraph(`Number of incomplete features: ${countNotDone}`);

```

<img width="662" alt="image" src="https://github.com/mdelobelle/metadatamenu/assets/3197629/5cb98060-c7f4-4968-886a-d5e9593d354a">

How can I adapt this code so that it output the "countNotDone" in a lookup field?

I'm really having trouble understanding how to output anything, even just pages.length for examples

**Comment by renekalff** [comment]
Would a formula field type work? Something like the following:
```javascript
(current.depends_on_features || []).filter(f => dv.page(f).feature_status !== "Completed").length
```
